deployments/epinio.go

```diff
@@ -167,8 +193,25 @@ func (k Epinio) Upgrade(c *kubernetes.Cluster, ui *termui.UI, options kubernetes
 }
 
 // Replaces ##current_epinio_version## with version.Version and applies the embedded yaml
-func (k Epinio) applyEpinioServerYaml(c *kubernetes.Cluster, ui *termui.UI) (string, error) {
-	yamlPathOnDisk, err := helpers.ExtractFile(epinioServerYaml)
+func (k Epinio) applyEpinioConfigYaml(c *kubernetes.Cluster, ui *termui.UI, user, pass string) (string, error) {
+	// (xxx) Apply traefik v2 middleware. This will fail for a
+	// traefik v1 controller.  Ignore error if it was due due to a
+	// missing Middleware CRD. That indicates presence of the
+	// traefik v1 controller. See also (yyy).
+
+	yamlPathOnDisk, err := helpers.ExtractFile(epinioBasicAuthYaml)
+	if err != nil {
+		return "", errors.New("Failed to extract embedded file: " + epinioBasicAuthYaml + " - " + err.Error())
+	}
+	defer os.Remove(yamlPathOnDisk)
+
+	out, err := helpers.Kubectl(fmt.Sprintf("apply -n %s --filename %s", EpinioDeploymentID, yamlPathOnDisk))
+	if err != nil && !strings.Contains(out, `no matches for kind "Middleware"`) {
```

Comment:
I wonder if different k8s versions could produce different error messages. If that happens we should probably stop supporting traefik v1 rather than trying to catch everything. Let's see.